xarray/backends/h5netcdf_.py

```diff
@@ -55,43 +86,63 @@ def __init__(self, filename, mode='r', format=None, group=None,
         opener = functools.partial(_open_h5netcdf_group, filename, mode=mode,
                                    group=group)
         self.ds = opener()
+        self._isopen = True
         self.format = format
         self._opener = opener
         self._filename = filename
         self._mode = mode
         super(H5NetCDFStore, self).__init__(writer)
 
+    @contextlib.contextmanager
+    def ensure_open(self, autoclose=True):
```

Comment:
@shoyer, the problem is that in order to avoid the open file limit we need to periodically open and close files.  Hence, we should open a file, read it, and then close it.  Otherwise during `open_mfdataset`, for instance, we will have too many open file and error.  However, you'll note that if we need to set properties we don't autoclose the files.  I've made this more explicit to make the code clearer by getting rid of the default `autoclose` argument.